src/components/Description/Description.tsx

```diff
@@ -0,0 +1,136 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import React from 'react';
+
+import { ReactNode, ComponentPropsWithoutRef, cloneElement } from 'react';
+import { TbTriangle, TbTriangleFilled } from 'react-icons/tb';
+
+export interface DescriptionProps extends ComponentPropsWithoutRef<'div'> {
+  preIcon: JSX.Element;
+  title: string;
+  website: string;
+  description: string;
+  votes?: number;
+  voted?: 0 | 1 | -1;
+  onVoteUp?: () => void;
+  onVoteDown?: () => void;
+  customClass?: string;
+  variant?: 'secondary' | 'primary' | undefined;
+  children?: ReactNode;
+}
+
+interface Classes {
+  [key: string]: string | object;
+}
+
+interface VoteProps extends ComponentPropsWithoutRef<'div'> {
+  votes: number;
+  voted: 0 | 1 | -1;
+  onVoteUp: () => void;
+  onVoteDown: () => void;
+}
+
+function Vote(props: VoteProps) {
+  const { votes, voted, onVoteUp, onVoteDown } = props;
+
+  function onClickItem(arg: 1 | -1) {
+    switch (voted) {
+      case 0: {
+        arg === 1 ? onVoteUp() : onVoteDown();
+        break;
+      }
+      case 1: {
+        arg === 1 ? onVoteDown() : (onVoteDown(), onVoteDown());
+        break;
+      }
+      case -1: {
+        arg === -1 ? onVoteUp() : (onVoteUp(), onVoteUp());
+        break;
+      }
+    }
+  }
```

Comment:
I think we can improve code comprehension here. Here's what I propose:

```suggestion
  function onClickItem(arg: 1 | -1) {
    switch (voted) {
      case VOTE: {
        arg === 1 ? onVoteDown() : (onVoteDown(), onVoteDown());
        break;
      }
      case UNVOTE: {
        arg === -1 ? onVoteUp() : (onVoteUp(), onVoteUp());
        break;
      }
      default: {
        arg === 1 ? onVoteUp() : onVoteDown();
        break;
      }
    }
  }
```

@mazmassa You know what I mean ? Or not ?

Reply:
```typescript
const VOTE = 1;
const UNVOTE = -1;
```

Reply:
done